paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -216,72 +438,305 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
+                                             memory::format::x);
+      mkldnninfo->conv_pd = ConvFwdPrimitiveDesc(
+          src_md, weights_md, bias_md, dst_md, *convinfo->strides,
+          *convinfo->paddings, *mkldnninfo->mkldnn_engine,
+          mkldnninfo->fuse_relu, mkldnninfo->fuse_residual_conn,
+          output_shift_scale, sum_scale, mkldnninfo->is_test);
+    } else {
+      mkldnninfo->conv_pd = ConvFwdPrimitiveDesc(
+          src_md, weights_md, dst_md, *convinfo->strides, *convinfo->paddings,
+          *mkldnninfo->mkldnn_engine, mkldnninfo->fuse_relu,
+          mkldnninfo->fuse_residual_conn, output_shift_scale, sum_scale,
+          mkldnninfo->is_test);
+    }
+    // Save conv_pd/src_memory/weights_memory for backward pass
+    dev_ctx->SetBlob(*mkldnninfo->key_conv_pd, mkldnninfo->conv_pd);
+
+    mkldnninfo->handler.reset(new platform::ConvMKLDNNHandler(
+        mkldnninfo->conv_pd, *dev_ctx, *mkldnninfo->mkldnn_engine,
+        *mkldnninfo->key));
+
+    // create mkldnn memory from input tensors (data/weights)
+    mkldnninfo->user_src_memory_p = mkldnninfo->handler->AcquireSrcMemory(
+        user_src_md, to_void_cast<T>(input_data));
+    auto user_weights_memory_p = mkldnninfo->handler->AcquireWeightsMemory(
+        user_weights_md, to_void_cast<K>(filter_data));
 
-      auto bias_memory_p =
-          handler.AcquireBiasMemoryFromPrimitive(user_bias_memory_p, pipeline);
-      conv_p = handler.AcquireConvolution(src_memory_p, weights_memory_p,
-                                          bias_memory_p, dst_memory_p);
+    // create reorder primitive if the input format is not the preferred one
+    mkldnninfo->src_memory_p =
+        mkldnninfo->handler->AcquireSrcMemoryFromPrimitive(
+            mkldnninfo->user_src_memory_p, *mkldnninfo->pipeline);
+
+    std::shared_ptr<mkldnn::memory> weights_memory_p;
+    int mask_reorder = is_multi_channel
+                           ? ((convinfo->g != 1) ? (1 << 1) + (1 << 0) : 1 << 0)
+                           : 0;
+    weights_memory_p = mkldnninfo->handler->AcquireWeightsMemoryFromPrimitive(
+        user_weights_memory_p, *mkldnninfo->pipeline, mkldnninfo->is_test,
+        is_INT8, scale_weights_data, mask_reorder);
+
+    if (mkldnninfo->fuse_residual_conn) {
+      auto residual_param = ctx.Input<Tensor>("ResidualData");
+      PADDLE_ENFORCE_EQ(output->dims(), residual_param->dims(),
+                        "Output and elementwise parameter need to have the "
+                        "same dimension sizes");
+      auto residual_dt =
+          paddle::framework::ToMKLDNNDataType(residual_param->type());
+      PADDLE_ENFORCE_EQ(
+          residual_param->format(), mkldnninfo->handler->GetDstFormat(),
+          "Conv input dimension and filter dimension should be the same.");
+      output->ShareDataWith(*residual_param);
+      if (residual_dt == mkldnn::memory::data_type::u8) {
```

Comment:
This
```C++
output_data = (...)
dst_memory_p = (..., output_data)
```
Seem to be a recurring pattern until line 665. Perhaps we could create a template function that would perform this action supplied with different variable types (`uint8_t, int8_t, float`)?